site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+            // MySpanishResponseConverter and MyResponseConverter will be used to make a response.
+            ...
+        }
+    }
+
+Armeria supports :ref:`media_type_nego`. So you may want to get a negotiated media type in order to set
+a ``Content-Type`` header on your response. In this case, you can access it in your response converter
+as follows.
+
+.. code-block:: java
+
+    class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx,
+                                            Object result) throws Exception {
+
+            MediaType mediaType = ctx.negotiatedProduceType();
+            if (mediaType != null) {
+                // Do something based on the media type.
+                ...
+            }
+            ...
+        }
+    }
+
+.. _more-about-converters-exception-handlers:
+
+More about converters and exception handlers
+--------------------------------------------
+
+There are 3 ways to specify converters and exception handlers, except that `@RequestConverter`_ annotates
+a variable. The first one is annotating a service method and the second one is annotating a service class,
+as you read the above sections. The last one is using ServerBuilder_ as follows.
+
+.. code-block:: java
+
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyExceptionHandler(), new MyRequestConverter(), new MyResponseConverter());
+
+When you specify exception handlers in every way like below, they are working in the following order commented.
+It is also the same as the applying order of the converters.
+
+.. code-block:: java
+
+    @ExceptionHandler(MyClassExceptionHandler3.class)           // order 3
+    @ExceptionHandler(MyClassExceptionHandler4.class)           // order 4
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @ExceptionHandler(MyMethodExceptionHandler1.class)      // order 1
+        @ExceptionHandler(MyMethodExceptionHandler2.class)      // order 2
+        public HttpResponse hello() { ... }
+    }
+
+    ...
+
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyGlobalExceptionHandler5(),        // order 5
+                        new MyGlobalExceptionHandler6());       // order 6
+
+Decorating an annotated service
+-------------------------------
+
+Every Service_ is able to be wrapped by another Service_ in Armeria (Refer to :ref:`server-decorator`
+for more information). Simply, you are able to write your own decorator by implementing
+DecoratingServiceFunction_ interface as follows.
+
+.. code-block:: java
+
+    class MyDecorator implements DecoratingServiceFunction<HttpRequest, HttpResponse> {
+        @Override
+        public HttpResponse serve(Service<HttpRequest, HttpResponse> delegate,
+                                  ServiceRequestContext ctx,
+                                  HttpRequest req) throws Exception {
```

Comment:
We could remove `throws Exception` whereever possible to reduce the cognitive load of a reader.